notebooks/Batchelor.ipynb

```diff
@@ -0,0 +1,348 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0c08f601-4df0-44f4-b0e6-ad1a9d545f06",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from dolfinx import *\n",
+    "from dolfinx import default_scalar_type\n",
+    "from dolfinx.fem import *\n",
+    "from mpi4py import *\n",
+    "from dolfinx.fem.petsc import LinearProblem\n",
+    "from dolfinx.plot import vtk_mesh\n",
+    "from petsc4py import PETSc\n",
+    "import numpy as np\n",
+    "import matplotlib\n",
+    "import matplotlib.pyplot as pl\n",
+    "import math\n",
+    "import pyvista\n",
+    "import ufl\n",
+    "import basix"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0bbac148-1658-4a4f-a1e9-57f5b77343ed",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# define exact solution, return it (ufl and numpy representation)\n",
+    "def u_exact_np(x):\n",
+    "    u = -1\n",
+    "    theta = np.arctan2(x[1],x[0])\n",
+    "\n",
+    "    d_psi_d_r = (u/((1/4)*(np.pi**2)-1))*((-1/4)*(np.pi**2)*(np.sin(theta))+((1/2)*(np.pi)*(theta)*(np.sin(theta)))+((theta)*(np.cos(theta))))\n",
+    "    d_psi_d_theta_overR = ((u)/((1/4)*(np.pi**2)-1)*(((-1/4)*(np.pi**2)*np.cos(theta))+((1/2)*(np.pi)*(np.sin(theta)))+((1/2)*(np.pi)*(theta)*(np.cos(theta)))+(np.cos(theta))-((theta)*(np.sin(theta)))))\n",
+    "\n",
+    "    return np.array([(((np.cos(theta) * (d_psi_d_theta_overR))) + (np.sin(theta) * d_psi_d_r)), (((np.sin(theta) * (d_psi_d_theta_overR))) - (np.cos(theta) * d_psi_d_r))])\n",
+    "\n",
+    "def u_exact(x):\n",
+    "    u = -1\n",
+    "    theta = ufl.atan2(x[1],x[0])\n",
+    "\n",
+    "    d_psi_d_r = (u/((1/4)*(ufl.pi**2)-1))*((-1/4)*(ufl.pi**2)*(ufl.sin(theta))+((1/2)*(ufl.pi)*(theta)*(ufl.sin(theta)))+((theta)*(ufl.cos(theta))))\n",
+    "    d_psi_d_theta_overR = ((u)/((1/4)*(ufl.pi**2)-1)*(((-1/4)*(ufl.pi**2)*ufl.cos(theta))+((1/2)*(ufl.pi)*(ufl.sin(theta)))+((1/2)*(ufl.pi)*(theta)*(ufl.cos(theta)))+(ufl.cos(theta))-((theta)*(ufl.sin(theta)))))\n",
+    "\n",
+    "    return ufl.as_vector([(((ufl.cos(theta) * (d_psi_d_theta_overR))) + (ufl.sin(theta) * d_psi_d_r)), (((ufl.sin(theta) * (d_psi_d_theta_overR))) - (ufl.cos(theta) * d_psi_d_r))])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4732c4cf-013f-4f2a-9071-85b408317528",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# define a function which returns values for Dirichlet boundary condition (60)\n",
+    "def u_0(x):\n",
+    "    u = 1\n",
+    "    us = u * np.ones_like(x[0])\n",
+    "    zeroes = np.zeros_like(x[0])\n",
+    "    \n",
+    "    return np.array([us, zeroes])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2d81ad31-0597-42d4-ad0e-ac770045eedb",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def batchelor(ne, p=1):\n",
```

Comment:
Could you rename this function to something like `solve_batchelor` to be consistent with the notation in the paper and the rest of the repository please?